### PR TITLE
CMake mac tweaks

### PR DESCRIPTION
**Pull request by danchr**
This PR contains a few minor fixes and tweaks for the CMake build infrastructure on a Mac, kept separate from a larger and possibly more controversial change.

### CODE REVIEW

File: cmake/Options.cmake [hunk 1]
```diff
@@ -46,6 +46,8 @@ function(set_options)
         set(DEFAULT_OPTION_INSTALL_FHS NO)
     endif (UNIX AND NOT APPLE)
 
+    option(OPTION_FORCE_COLORED_OUTPUT "Always produce ANSI-colored output (GNU/Clang only)." FALSE)
```
Comment (LordAro) [inline cmake/Options.cmake]:
```suggestion
    option(OPTION_FORCE_COLORED_OUTPUT "Always produce ANSI-colored output (GNU/Clang only)." NO)
```
For consistency

File: cmake/CompileFlags.cmake [hunk 2]
```diff
@@ -68,14 +68,29 @@ macro(compile_flags)
...
+        # When we are a stable release (Release build + USE_ASSERTS not set),
+        # assertations are off, which trigger a lot of warnings. We disable
+        # these warnings for these releases.
+        if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+            add_compile_options(
+                "$<${IS_STABLE_RELEASE}:-Wno-unused-variable>"
+                "$<${IS_STABLE_RELEASE}:-Wno-unused-but-set-parameter>"
+                "$<${IS_STABLE_RELEASE}:-Wno-unused-but-set-variable>"
+                )
+        else(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
```
Comment (LordAro) [inline cmake/CompileFlags.cmake]: Make sure alignment & spaces is consistent

File: cmake/CompileFlags.cmake [hunk 3]
```diff
@@ -68,14 +68,29 @@ macro(compile_flags)
...
+            )
+        endif (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+
+        if (${OPTION_FORCE_COLORED_OUTPUT})
+            if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+                add_compile_options (-fdiagnostics-color=always)
+            elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang" OR CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang")
+                add_compile_options (-fcolor-diagnostics)
+            endif ()
+        endif ()
```
Comment (LordAro) [inline cmake/CompileFlags.cmake]: Currently, we're putting the condition in the endif
(Personally I'd like to remove these, but for now, consistency)

File: CMakeLists.txt [hunk 4]
```diff
@@ -33,7 +34,9 @@ if (NOT WIN32)
     find_package(Fluidsynth)
     find_package(Freetype)
     find_package(Fontconfig)
-    find_package(ICU OPTIONAL_COMPONENTS i18n lx)
+    if (NOT APPLE)
+      find_package(ICU OPTIONAL_COMPONENTS i18n lx)
```
Comment (LordAro) [inline CMakeLists.txt]: Have you also checked the various source lists for icu-specific files that should be excluded?

Comment (LordAro) [inline CMakeLists.txt]: also, 4 spaces for indentation (also below)

**Review by LordAro — CHANGES_REQUESTED** [review]
Nothing major

File: cmake/Options.cmake (hunk 1 above)
Comment (danchr) [reply cmake/Options.cmake]: Fixed — along with another reference elsewhere 🙂

File: CMakeLists.txt (hunk 4 above)
Comment (danchr) [reply CMakeLists.txt]: Well, I'm not _that_ much into how OpenTTD actually works, but these two seemed safe.

Comment (danchr) [reply CMakeLists.txt]: Fixed.

**Review by LordAro — DISMISSED** [review]

File: CMakeLists.txt [hunk 5]
```diff
@@ -161,8 +164,10 @@ if (NOT OPTION_DEDICATED)
     link_package(Allegro)
     link_package(FREETYPE TARGET Freetype::Freetype)
     link_package(Fontconfig TARGET Fontconfig::Fontconfig)
-    link_package(ICU_lx)
-    link_package(ICU_i18n)
+    if (NOT APPLE)
+        link_package(ICU_lx)
+        link_package(ICU_i18n)
+    endif (NOT APPLE)
```
Comment (glx22) [inline CMakeLists.txt]: No need for `NOT APPLE` here, the package is not found anyway

Comment (danchr) [reply CMakeLists.txt]: Well, without this change, the package is found if present; I just checked and `icu-lx` does indeed get linked.

Comment (glx22) [reply CMakeLists.txt]: But find_package() (line 38) is guarded so it's not even searched. Be sure to always clear the cache when testing modifications in cmake files.

Comment (danchr) [reply CMakeLists.txt]: Ah, I see — I thought you were referring to the _other_, vaguely similar line. I've removed this one!

File: cmake/CompileFlags.cmake [hunk 6]
```diff
@@ -68,14 +68,29 @@ macro(compile_flags)
...
+                "$<${IS_STABLE_RELEASE}:-Wno-unused-but-set-variable>"
+            )
+        else (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+            add_compile_options(
+                "$<${IS_STABLE_RELEASE}:-Wno-unused-variable>"
+                "$<${IS_STABLE_RELEASE}:-Wno-unused-parameter>"
+            )
+        endif (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+
+        if (${OPTION_FORCE_COLORED_OUTPUT})
```
Comment (glx22) [inline cmake/CompileFlags.cmake]: `if (OPTION_FORCE_COLORED_OUTPUT)` should work here

Comment (LordAro) [reply cmake/CompileFlags.cmake]: That said, why add it at all? Why is it "useful for Ninja" ? Big ol comment somewhere please

Comment (danchr) [reply cmake/CompileFlags.cmake]: > That said, why add it at all? Why is it "useful for Ninja" ? Big ol comment somewhere please

Ninja processes the output so the output from the compiler isn't directly to a terminal; hence, the default is non-coloured output.

Comment (LordAro) [reply cmake/CompileFlags.cmake]: I figured that would be the reason, but I'd like to see a comment in the code explaining it

File: cmake/Options.cmake [hunk 7]
```diff
@@ -46,6 +46,8 @@ function(set_options)
         set(DEFAULT_OPTION_INSTALL_FHS NO)
     endif (UNIX AND NOT APPLE)
 
+    option(OPTION_FORCE_COLORED_OUTPUT "Always produce ANSI-colored output (GNU/Clang only)." NO)
```
Comment (glx22) [inline cmake/Options.cmake]: Option values should be ON/OFF.

Comment (LordAro) [reply cmake/Options.cmake]: That change is presumably the reason for the conflict

File: cmake/CompileFlags.cmake (hunk 6 above)
Comment (danchr) [reply cmake/CompileFlags.cmake]: Fixed!

File: cmake/Options.cmake (hunk 7 above)
Comment (danchr) [reply cmake/Options.cmake]: Fixed!

**Comment by danchr** [comment]
I applied the requested changes and rebased onto `master`; I also added a separate commit to suppress checking for SDL and FreeType on macOS.

**Review by LordAro — APPROVED** [review]

**Pull request merged**